src/Magento/FunctionalTestingFramework/Test/Objects/ActionObject.php

```diff
@@ -417,6 +425,22 @@ private function resolveUrlReference()
         }
     }
 
+    /**
+     * Returns array of missing references
+     *
+     * @param string $replacement
+     * @return array
+     */
+    private function getMissingReferences($replacement): array
+    {
+        $mustachePattern = '/({{[\w]+}})|({{[\w]+\.[\w\[\]]+}})|({{[\w]+\.[\w]+\((?(?!}}).)+\)}})/';
```

Comment:
For readability's sake I think we should split this pattern into smaller variables like:
```
//match {{somepage}}
$simpleGroup = '({{[\w]+}})'
//match {{field.key}} or {{field.key[1]}}
$fullGroup = <...>
```

And bring the all into a final `$regexPattern`. It may create some small vertical clutter, but I think the gains in readability outweigh the downside of verbosity.

Reply:
I don't like your approach, so introduced mine.
One day when we will refactor whole MFTF - I'm going to use existing tools to extract the Mustache patterns :-)

Reply:
Your approach looks good as well, gets the spirit of the suggestion 👍